mock-mhs-adaptor/src/main/java/uk/nhs/adaptors/mockmhsservice/service/MockMhsService.java

```diff
@@ -0,0 +1,83 @@
+package uk.nhs.adaptors.mockmhsservice.service;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.jms.JmsException;
+import org.springframework.web.bind.annotation.RestController;
+import io.micrometer.core.instrument.util.IOUtils;
+import lombok.RequiredArgsConstructor;
+
+import static org.springframework.http.HttpStatus.ACCEPTED;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+
+import uk.nhs.adaptors.mockmhsservice.common.InboundMessage;
+import uk.nhs.adaptors.mockmhsservice.producer.InboundProducer;
+
+@RestController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class MockMhsService {
+
+    private final ObjectMapper objectMapper;
+    private final InboundProducer inboundProducer;
+
+    public ResponseEntity<String> handleRequest(
+            String interactionId,
+            String waitForResponse,
+            String fromAsid,
+            String messageId,
+            String correlationId,
+            String odsCode,
+            String mockMhsMessage) throws IOException {
```

Comment:
unused vars 

String waitForResponse,
            String fromAsid,
            String messageId,
            String correlationId,
            String odsCode,

Reply:
Wrap in an entity as @Broom94 suggested in the controller